test/schemas/testpilot.js

```diff
@@ -44,154 +55,43 @@ describe("The 'testpilot' Schema", function() {
-  });
-  it("should reject if required field 'timestamp' is missing", function(done) {
-    const ping = clone(samplePing);
-    delete ping.payload.timestamp;
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required field 'timestamp' is a Date object", function(done) {
-    const ping = clone(samplePing);
-    ping.payload.timestamp = new Date();
-    ping.payload.events[0].timestamp = new Date().toISOString();
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required field 'timestamp' is a Date ISO string", function(done) {
-    const ping = clone(samplePing);
-    ping.payload.timestamp = new Date().toISOString();
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required field 'version' is missing", function(done) {
-    const ping = clone(samplePing);
-    delete ping.payload.version;
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required field 'version' is malformed", function(done) {
-    const ping = clone(samplePing);
-    ping.payload.version = 12345; // version is an AMO version string, not an integer
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required event field 'event' is missing", function(done) {
-    const ping = clone(samplePing);
-    delete ping.payload.events[0].event;
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required event field 'event' is malformed", function(done) {
-    const ping = clone(samplePing);
-    ping.payload.events[0].event = 1;
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required event field 'object' is missing", function(done) {
+  it("should reject if required field topic is incorrect", function(done) {
     const ping = clone(samplePing);
-    delete ping.payload.events[0].object;
+    ping.topic = "foo";
     Joi.validate(ping, testPilotSchema, function(err) {
       expect(err).to.exist;
       done();
     });
   });
-  it("should reject if required event field 'object' is malformed", function(done) {
-    const ping = clone(samplePing);
-    ping.payload.events[0].object = 1;
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
+  requiredStrings.forEach((field) => {
+    it(`should reject if required field ${field} is missing`, function(done) {
+      const ping = clone(samplePing);
+      delete ping[field];
+      Joi.validate(ping, testPilotSchema, function(err) {
+        expect(err).to.exist;
```

Comment:
I think we're just asserting that the `err` is non-null here. Do we also want to assert that we're receiving an expected `err.message` or something so that we know we're getting an expected error and not something unrelated?

(I'll also happily accept "shut up, that's unrelated to this PR" as a valid response)